pkg/debuggers/client.go

```diff
@@ -32,10 +33,24 @@ func RunSquashClient(debugger func(string) Debugger, conttopid platforms.Contain
 		return err
 	}
 
-	// TODO(mitchdraft) - the debug handler will need to be spawned/removed when a new debugging session is opened/closed
-	// The debug initialization request will include the watchNamespace. For now, use a known placeholder
-	watchNamespace := options.SquashClientNamespace
-	return NewDebugHandler(ctx, watchNamespace, daClient, debugger, conttopid).handleAttachments()
+	var kubeResClient *kubernetes.Clientset
+	if inClusterMode {
+		kubeResClient, err = kubeutils.NewInClusterKubeClientset()
+		if err != nil {
+			return err
+		}
+	} else {
+		kubeResClient, err = kubeutils.NewOutOfClusterKubeClientset()
+		if err != nil {
+			return err
+		}
+	}
+	watchNamespaces, err := kubeutils.GetNamespaces(kubeResClient)
```

Comment:
this will only watch current namespaces (will not watch for namespaces that are created later)
right now this is needed as solo-kit doesn't support watching all namespaces. 
add a todo to fix this once solo-kit supports this

Reply:
made an issue for this